integration/spark/src/main/scala/org/apache/spark/sql/execution/command/mutation/CarbonProjectForUpdateCommand.scala

```diff
@@ -268,6 +267,11 @@ private[sql] case class CarbonProjectForUpdateCommand(
         // When the table has too many segemnts, it will take a long time.
```

Comment:
how about separate update and horizonCompaction.
if horizonCompaction failed, the update may be already successful.

Reply:
I have modified code according to your suggestion